### PR TITLE
Add TonPools

### PR DESCRIPTION
**Pull request by nuhbye**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/tonpools exports TVL: 
        
 
 
    ton                       2.15 k
    
    total                    2.15 k

File: projects/tonpools/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const { call } = require("../helper/chain/ton");
+
+module.exports = {
+  timetravel: false,
+  ton: {
+    tvl: async () => {
+      const result = await call({ target: "EQA7y9QkiP4xtX_BhOpY4xgVlLM7LPcYUA4QhBHhFZeL4fTa", abi: "getActiveContest"});
```
Comment (g1nt0ki) [inline projects/tonpools/index.js:7]: hi, why is this value higher than amount on ton in the contract?

Comment (g1nt0ki) [reply projects/tonpools/index.js:7]: sorry, we can count only the value of tokens on the contract. I thought when user joins the lottery, the TON is deposited into this contract, is that assumption wrong?

Comment (nuhbye) [reply projects/tonpools/index.js:7]: Yes, the user tokens are deposited in the contract only. Also, when a winner is decided the winnings are added to the contract for user to claim whenever they want. Both the winnings and user tokens deposited contribute to the total tokens in the contract.

**Comment by llamatester** [comment]
The adapter at projects/tonpools exports TVL: 
        
 
 
    ton                       5.32 k
    
    total                    5.32 k

File: projects/tonpools/index.js (hunk 1 above)
Comment (nuhbye) [reply projects/tonpools/index.js]: I have updated the code now to just pick the contract's token balance, I hope this should work.

**Comment by Define101** [comment]
@nuhbye website link please

**Comment by nuhbye** [comment]
Here is the link to the website: https://www.tonpools.com/

**Comment by nuhbye** [comment]
Also, any idea on whom can I reach out for getting https://github.com/DefiLlama/defillama-server/pull/7957 approved as well. I guess this is also required to get it listed.

**Pull request merged**